src/main/java/com/mindee/parsing/common/ocr/OcrPage.java

```diff
@@ -0,0 +1,78 @@
+package com.mindee.parsing.common.ocr;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mindee.geometry.PolygonUtils;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+/**
+ * Represents a page.
+ */
+@Getter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class OcrPage {
+
+  @JsonProperty("all_words")
+  private ArrayList<Word> words = new ArrayList<>();
+
+  /**
+   * Order all the words on the page into lines.
+   */
+  public ArrayList<ArrayList<Word>> toLines() {
+    Word current = null;
+    ArrayList<Integer> indexes = new ArrayList<>();
+    ArrayList<ArrayList<Word>> lines = new ArrayList<>();
+
+    this.words.sort((word1, word2) -> PolygonUtils.CompareOnY(word1.getPolygon(), word2.getPolygon()));
```

Comment:
This sort causes the toLines method to modify the this.words List - it can lead to some unexpected behaviour from the users perspective. For instance : 

Word ocrWords1 = ocr.getWords().get(0);
ocr.toString();
Word ocrWords2 = ocr.getWords().get(0);

assert ocrWords1.equals(ocrwords2);  // this may fail

We can consider running toLines() once during construction, so that the words list is always sorted. Or we can keep the words list as is, and sort using a local variable.

Reply:
I've added a unit test to verify this but it never comes up as different, even when playing around and forcing it to be different. It does not look like the word order matters :sweat: 

Can you take a look here and see what I'm doing wrong?
https://github.com/mindee/mindee-api-java/pull/102/files#diff-69a95c97d1ad4fc78d442944e71eabaa5449b955ed3e93d20d00e311223542e8R33

I would like to have the test failing before making changes to the code logic.

Reply:
I'm getting the same result in the test when changing:

`ocr.getMVisionV1().getPages().get(0).getAllLines();`

to:

`ocr.toString();`

(the `toString` method internally calls `getAllLines` which itself calls `toLines`)

Reply:
Awesome, thanks!